sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -315,9 +315,28 @@ object GpuDeviceManager extends Logging {
       }
 
       Cuda.setDevice(gpuId)
-      Rmm.initialize(init, logConf, poolAllocation)
-      RapidsBufferCatalog.init(conf)
+      try {
+        Rmm.initialize(init, logConf, poolAllocation)
+      } catch {
+        case firstEx: CudfException if ((init | RmmAllocationMode.CUDA_ASYNC) != 0) => {
+          logWarning("Failed to initialize RMM with ASYNC allocator. " +
+            "Try to initialize it again with ARENA allocator as a fallback option.")
+          init = init & (~RmmAllocationMode.CUDA_ASYNC) | RmmAllocationMode.ARENA
+          try {
+            Rmm.initialize(init, logConf, poolAllocation)
+          } catch {
+            case secondEx: Throwable => {
+              logError(
+                "Failed to initialize RMM with either ASYNC or ARENA allocators. Exiting...")
+              secondEx.addSuppressed(firstEx)
+              throw secondEx
+            }
+          }
+        }
+        case ex: Throwable => throw ex
```

Comment:
This is not necessary. Scala will throw out the exceptions been not caught explicitly by case statements.